catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -14,332 +14,336 @@
                         source: this.model,
-                        mode: 'add'
+                        mode: 'edit'
                     })
                 );
-            }
-        }
-    });
-
-    SourceView.ModalController = Marionette.Controller.extend({
-        initialize: function (options) {
-            this.application = options.application;
-            this.listenTo(wreqr.vent, "showModal", this.showModal);
-        },
-        showModal: function(modalView) {
-            // Global div for workaround with iframe resize and modals
-            var region = this.application.getRegion('sourcesModal');
-            var collectionRegion = this.application.getRegion('collectionRegion');
-            var iFrameModalDOM = $('#IframeModalDOM');
-            modalView.$el.on('hidden.bs.modal', function () {
-                iFrameModalDOM.hide();
-            });
-            modalView.$el.on('shown.bs.modal', function () {
-                var extraHeight = modalView.el.firstChild.clientHeight - collectionRegion.$el.height();
-                if(extraHeight > 0) {
-                    iFrameModalDOM.height(extraHeight);
-                    iFrameModalDOM.show();
+            },
+            removeSource: function () {
+                if (this.model) {
+                    wreqr.vent.trigger("showModal",
+                        new SourceView.DeleteModal({
+                            model: this.model,
+                            collection: this.model.get('collection')
+                        })
+                    );
                 }
-            });
-            region.show(modalView);
-            region.currentView.$el.modal();
-        }
-    });
+            },
+            addSource: function () {
+                if (this.model) {
+                    wreqr.vent.trigger("showModal",
+                        new ModalSource.View({
+                            model: this.model.getSourceModelWithServices(),
+                            source: this.model,
+                            mode: 'add'
+                        })
+                    );
+                }
+            }
+        });
 
-    SourceView.DeleteItem = Marionette.ItemView.extend({
-        template: "deleteSource"
-    });
+        SourceView.ModalController = Marionette.Controller.extend({
+            initialize: function (options) {
+                this.application = options.application;
+                this.listenTo(wreqr.vent, "showModal", this.showModal);
+            },
+            showModal: function (modalView) {
+                // Global div for workaround with iframe resize and modals
+                var region = this.application.getRegion('sourcesModal');
+                var collectionRegion = this.application.getRegion('collectionRegion');
+                var iFrameModalDOM = $('#IframeModalDOM');
+                modalView.$el.on('hidden.bs.modal', function () {
```

Comment:
These events are never removed. This is going to turn each modalView into a zombie.